### PR TITLE
feat(preview): Focus on comment form for replies and modifying

### PR DESCRIPTION
**Pull request by ahorowitz123**
- Used draft.js functionality in `EditorState` to force focus on the comment form when it is open from either initiating a new reply or modifying and existing comment/reply
- Added `shouldFocusOnOpen` prop to `CommentForm` which allows callers to opt in to this behavior

*Note regarding tests* -- While it would be better to simulate click events and check browser focus of the textbox element, the `EditorState.moveFocusToEnd` uses a lot of window functions which don't exist in our current version of jsdom. I tried upgrading jsdom which solved the issue but focusing on the text box still wasn't working correctly. Given that this function is part of draft.js, I opted to mock it and check that was called.

focus on comment form

### CODE REVIEW

**Review by JChan106 — COMMENTED** [review]
Have we checked if this works with multiple comments and opening the reply form between them to make sure the focus switches properly?

File: src/elements/content-sidebar/activity-feed/comment/BaseComment.js [hunk 1]
```diff
@@ -302,22 +302,23 @@ export const BaseComment = ({
...
-                            isEditing={isEditing}
-                            entityId={id}
+                            shouldFocusOnOpen
                             tagged_message={tagged_message}
-                            getAvatarUrl={getAvatarUrl}
-                            mentionSelectorContacts={mentionSelectorContacts}
-                            getMentionWithQuery={getMentionWithQuery}
+                            updateComment={handleMessageUpdate}
+                            // $FlowFixMe
+                            user={currentUser}
```
Comment (jcamden) [inline src/elements/content-sidebar/activity-feed/comment/BaseComment.js:321]: +1 for alphabetized props. Support it, but it is admittedly a little hard to read the diff. I can't see any changes, but I just wanted to confirm that.

File: src/elements/content-sidebar/activity-feed/comment/CreateReply.js [hunk 2]
```diff
@@ -49,15 +49,16 @@ const CreateReply = ({
...
-                    showTip={false}
+                    mentionSelectorContacts={mentionSelectorContacts}
                     onCancel={onCancel}
                     onFocus={onFocus}
-                    createComment={handleSubmit}
-                    mentionSelectorContacts={mentionSelectorContacts}
-                    getMentionWithQuery={getMentionWithQuery}
                     placeholder={placeholder}
+                    shouldFocusOnOpen
+                    showTip={false}
```
Comment (jcamden) [inline src/elements/content-sidebar/activity-feed/comment/CreateReply.js:61]: Same here: just alphabetization?

File: src/elements/content-sidebar/activity-feed/comment/__tests__/BaseComment.test.js [hunk 3]
```diff
@@ -55,6 +55,10 @@ export const getWrapper = props =>
     );
 
 describe('elements/content-sidebar/ActivityFeed/comment/BaseComment', () => {
+    beforeEach(() => {
+        EditorState.moveFocusToEnd = editor => editor;
```
Comment (jcamden) [inline src/elements/content-sidebar/activity-feed/comment/__tests__/BaseComment.test.js:59]: Wh... wha? Just a function that returns its arg? Is this how you are clearing mocks in between tests?

**Review by jcamden — APPROVED** [review]
Looks good! Left some inconsequential comments.

File: src/elements/content-sidebar/activity-feed/comment-form/__tests__/CommentForm.test.js [hunk 4]
```diff
@@ -12,15 +14,13 @@ const intlFake = {
...
-                {...props}
-            />,
-        );
+    const getComponent = (props = {}) => (
+        <CommentForm getMentionWithQuery={() => {}} intl={intlFake} user={{ id: 123, name: 'foo bar' }} {...props} />
+    );
+
+    const render = props => mount(getComponent(props));
+
+    const renderRTL = props => renderRTLFunc(getComponent(props));
```
[the remark this answers is not in the document]
Comment (ahorowitz123) [reply src/elements/content-sidebar/activity-feed/comment-form/__tests__/CommentForm.test.js:23]: Good call. I just copied what was already there but I'll update this

File: src/elements/content-sidebar/activity-feed/comment/CreateReply.js (hunk 2 above)
Comment (ahorowitz123) [reply src/elements/content-sidebar/activity-feed/comment/CreateReply.js:61]: Same as comment above

File: src/elements/content-sidebar/activity-feed/comment/BaseComment.js (hunk 1 above)
Comment (ahorowitz123) [reply src/elements/content-sidebar/activity-feed/comment/BaseComment.js:321]: I only added `shouldFocusOnOpen`. The rest is just alphabetizing.

File: src/elements/content-sidebar/activity-feed/comment/__tests__/BaseComment.test.js (hunk 3 above)
Comment (ahorowitz123) [reply src/elements/content-sidebar/activity-feed/comment/__tests__/BaseComment.test.js:59]: I guess I don't need to set this before every test and can just do it once. Good call.

But yeah this is how I'm mocking it. It needs to return an editor to get other tests to work. But also breaks in jsdom if we try to actually execute the real function

**Comment by ahorowitz123** [comment]
> Have we checked if this works with multiple comments and opening the reply form between them to make sure the focus switches properly?

Yes since the function is called on component mount, every time a new component is mounted it gets focus

**Review by jcamden — APPROVED** [review]

**Review by joshmarnold — APPROVED** [review]

**Pull request merged**